### PR TITLE
Add another JSONEncoder.makeWithDefaults variation

### PR DESCRIPTION
**Pull request by yim-lee**
It would be nice if we can have sorted keys without pretty printing.

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test self hosted

File: Sources/Basics/JSON+Extensions.swift [hunk 1]
```diff
@@ -78,23 +78,35 @@ extension JSONDecoder {
 
 extension JSONEncoder {
     public static func makeWithDefaults(prettified: Bool = true, dateEncodingStrategy: DateEncodingStrategy = .safeISO8601) -> JSONEncoder {
+        Self.makeWithDefaults(sortKeys: prettified, prettyPrint: prettified, escapeSlashes: !prettified, dateEncodingStrategy: dateEncodingStrategy)
```
Comment (abertelrud) [inline Sources/Basics/JSON+Extensions.swift:81]: These defaults make sense to me based on `prettified`.  That makes for a convenient shorthand, I think.

**Review by abertelrud — APPROVED** [review]

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
interesting build failure here, unrelated to the PR

**Review by tomerd — APPROVED** [review]

**Comment by yim-lee** [comment]
The test failures are legit. I forgot to set encoder's outputFormatting. 🤦🏻‍♀️

@swift-ci please smoke test

**Pull request merged**